apps/TideForker/main.cpp

```diff
@@ -63,14 +63,25 @@ int main(int argc, char* argv[])
     COMMAND_LINE_PARSER_CHECK(CommandLineHelper, "tideForker");
 
     {
-        MPIChannelPtr worldChannel(new MPIChannel(argc, argv));
-        const int rank = worldChannel->getRank();
-        MPIChannelPtr localChannel(new MPIChannel(*worldChannel, 0, rank));
-        MPIChannelPtr localChannel2(new MPIChannel(*worldChannel, 0, rank));
-        MPIChannelPtr mainChannel(new MPIChannel(*worldChannel, 0, rank));
+        auto worldComm = MPICommunicator{argc, argv};
+        if (worldComm.getSize() < 2)
+        {
+            std::cerr << "MPI group size < 2 detected. Use tide script or check"
+                         " MPI parameters."
+                      << std::endl;
+            return EXIT_FAILURE;
+        }
+        auto masterForkerComm = MPICommunicator{worldComm, 0};
```

Comment:
Maybe add a comment explaining why you create these communicators which seem to be doing nothing.

Reply:
OK, will do